docs/source/x_ext.rst

```diff
@@ -139,6 +139,120 @@ before a ``WFI`` instruction have fully completed (so that sleep mode can be ent
 In short: From a functional perspective it should not matter whether an instruction is handled inside the core or inside a |coprocessor|. In both cases
 the instructions need to obey the same instruction dependency rules, memory consistency rules, load/store address checks, fences, etc.
 
+Interfaces
+----------
+
+This section describes the six interfaces of CORE-V-XIF. Port directions are described as seen from the perspective of the |processor|.
+The |coprocessor| will have opposite pin directions.
+Stated signals names are not mandatory, but it is highly recommended to at least include the stated names as part of actual signal names. It is for example allowed to add prefixes and/or postfixes (e.g. ``x_`` prefix or ``_i``, ``_o`` postfixes) or to use different capitalization. A name mapping should be provided if non obvious renaming is applied.
+
+SystemVerilog example
+~~~~~~~~~~~~~~~~~~~~~
+The description in this specification is based on SystemVerilog interfaces. Of course the use of SystemVerilog (interfaces) is not mandatory.
+
+A |processor| using the eXtension interface could have the following interface:
+
+.. code-block:: verilog
+
+  module cpu;
+  (
+    // eXtension interface
+    if_xif.cpu_compressed       xif_compressed_if,
+    if_xif.cpu_issue            xif_issue_if,
+    if_xif.cpu_commit           xif_commit_if,
+    if_xif.cpu_mem              xif_mem_if,
+    if_xif.cpu_mem_result       xif_mem_result_if,
+    if_xif.cpu_result           xif_result_if,
+
+    ... // Other ports omitted
+  );
+
+A full example of a |processor| with an eXtension interface is the **CV32E40X**, which can be found at https://github.com/openhwgroup/cv32e40x. 
+
+A |coprocessor| using the eXtension interface could have the following interface:
+
+.. code-block:: verilog
+
+  module coproc;
+  (
+    // eXtension interface
+    if_xif.coproc_compressed    xif_compressed_if,
+    if_xif.coproc_issue         xif_issue_if,
+    if_xif.coproc_commit        xif_commit_if,
+    if_xif.coproc_mem           xif_mem_if,
+    if_xif.coproc_mem_result    xif_mem_result_if,
+    if_xif.coproc_result        xif_result_if,
+
+    ... // Other ports omitted
+  );
+
+A SystemVerilog interface implementation for CORE-V-XIF could look as follows:
+
+.. code-block:: verilog
+
+  interface if_xif;
+  #(
+    parameter int          X_NUM_RS        =  2,  // Number of register file read ports that can be used by the eXtension interface
+    parameter int          X_MEM_WIDTH     =  32, // Memory access width for loads/stores via the eXtension interface
+    parameter int          X_RFR_WIDTH     =  32, // Register file read access width for the eXtension interface
+    parameter int          X_RFW_WIDTH     =  32, // Register file write access width for the eXtension interface
+    parameter logic [31:0] X_MISA          =  '0  // MISA extensions implemented on the eXtension interface
```

Comment:
The parameters of the interface are not complete. Namely the X_DATA_WIDTH, X_NUM_FRS, X_ID_WIDTH are missing.

Additionally, I am not sure why x_ was omitted for all signals, but not for the parameters.

Reply:
I will add them (they were not there because in the 40X we defined them elsewhere, but I agree they should be here).

Reply:
The x_ was removed from the signals because they would be prefixed by for example xif_issue_if. anyway, so repeating the x_ just made the names longer. For the parameters I want to connect them to the 40X top level parameters without name change and at the top level I want a clear X_ or xif_ prefix to mark them X interface related.